docs/notebooks/topic_coherence_model_selection.ipynb

```diff
@@ -394,109 +520,3275 @@
-      "Avg coherence for num_topics=40: 0.51029\n",
-      "Avg coherence for num_topics=80: 0.51147\n",
-      "Avg coherence for num_topics=50: 0.51582\n",
-      "Avg coherence for num_topics=20: 0.49602\n",
-      "Avg coherence for num_topics=90: 0.47067\n",
-      "Avg coherence for num_topics=60: 0.48913\n",
-      "Avg coherence for num_topics=30: 0.48709\n",
-      "CPU times: user 2min 39s, sys: 524 ms, total: 2min 39s\n",
-      "Wall time: 2min 40s\n"
+      "CPU times: user 3min 5s, sys: 8.26 s, total: 3min 13s\n",
+      "Wall time: 3min 15s\n"
      ]
     }
    ],
    "source": [
     "%%time\n",
-    "import numpy as np\n",
-    "# Next we perform the coherence evaluation for each of the models.\n",
-    "# Since we have already precomputed the probabilities, this simply\n",
-    "# involves using the accumulated stats in the `CoherenceModel` to\n",
-    "# perform the evaluations, which should be pretty quick.\n",
-    "\n",
-    "coherences = {}\n",
-    "for num_topics, topics in model_topics.items():\n",
-    "    cm.topics = topics\n",
-    "\n",
-    "    # We evaluate at various values of N and average them. This is a more robust,\n",
-    "    # according to: http://people.eng.unimelb.edu.au/tbaldwin/pubs/naacl2016.pdf\n",
-    "    coherence_at_n = {}\n",
-    "    for n in (20, 15, 10, 5):\n",
-    "        cm.topn = n\n",
-    "        topic_coherences = cm.get_coherence_per_topic()\n",
-    "        \n",
-    "        # Let's record the coherences for each topic, as well as the aggregated\n",
-    "        # coherence across all of the topics.\n",
-    "        coherence_at_n[n] = (topic_coherences, cm.aggregate_measures(topic_coherences))\n",
-    "        \n",
-    "    topic_coherences, avg_coherences = zip(*coherence_at_n.values())\n",
-    "    avg_topic_coherences = np.vstack(topic_coherences).mean(0)\n",
-    "    avg_coherence = np.mean(avg_coherences)\n",
-    "    print(\"Avg coherence for num_topics=%d: %.5f\" % (num_topics, avg_coherence))\n",
-    "    coherences[num_topics] = (avg_topic_coherences, avg_coherence)"
+    "\n",
+    "models_dir = os.path.join(home, 'workshop', 'nlp', 'models')\n",
+    "vectors_path = os.path.join(models_dir, 'GoogleNews-vectors-negative300.bin.gz')\n",
+    "keyed_vectors = models.KeyedVectors.load_word2vec_format(vectors_path, binary=True)\n",
+    "\n",
+    "# still need to estimate_probabilities, but corpus is not scanned\n",
+    "cm = models.CoherenceModel.for_models(\n",
+    "    trained_models.values(), dictionary, texts=corpus.get_texts(),\n",
+    "    coherence='c_w2v', keyed_vectors=keyed_vectors)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 68,
+   "execution_count": 33,
    "metadata": {
     "collapsed": false
    },
    "outputs": [
     {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:gensim.topic_coherence.indirect_confirmation_measure:at least one topic word not in word2vec model\n",
```

Comment:
Please suppress this warning here (and below)

Reply:
done